vumi/persist/fields.py

```diff
@@ -729,12 +729,18 @@ def setup(self, model_cls):
             self.index_name = "%s_bin" % self.key
         else:
             self.index_name = self.field.index
-        if self.field.backlink is None:
-            self.backlink_name = model_cls.__name__.lower() + "s"
-        else:
-            self.backlink_name = self.field.backlink
+
+        backlink_name = self.field.backlink
+        if backlink_name is None:
+            backlink_name = model_cls.__name__.lower() + "s"
         self.other_model.backlinks.declare_backlink(
-            self.backlink_name, self.reverse_lookup_keys)
+            backlink_name, self.reverse_lookup_keys)
+
+        backlink_keys_name = backlink_name + "_keys"
+        if backlink_keys_name.endswith("s_keys"):
+            backlink_keys_name = backlink_name[:-1] + "_keys"
```

Comment:
I would've thought it is as simple enough to avoid:

```python
if self.field.backlink is None:
    name = model_cls.__name__.lower()
    backlink_name = name + "s"
    backlink_keys_name = name + "_keys"
else:
    backlink_name = self.field.backlink
    backlink_keys_name = self.field.backlink + "_keys"

self.other_model.backlinks.declare_backlink(
    backlink_name, self.reverse_lookup_keys)

self.other_model.backlinks.declare_backlink(
    backlink_keys_name, self.reverse_lookup_keys_paginated)
```

What is the rabbit hole? Are we doing this somewhere else too?